src/Symfony/Component/TypeInfo/Type.php

```diff
@@ -86,4 +110,28 @@ private function testIdentifier(callable $test): bool
 
         return $this->is($callable);
     }
+
+    /**
+     * BC layer for Symfony\Component\PropertyInfo\Type.
+     *
+     * Can be removed as soon as Symfony\Component\PropertyInfo\Type is removed (8.0).
+     *
+     * @internal
+     */
+    public function setCollection(bool $collection): void
```

Comment:
`@internal` means code can only be used at *this* component level, so this method must not be called from PropertyInfo. 
Ideally TypeInfo should not be aware of PropertyInfo at all, not even temporarily. Can we somehow manage to keep PropertyInfo's specific code inside PropertyInfo?